middle.mkd

```diff
@@ -274,8 +274,9 @@ A GeoJSON object with the type "Feature" is a feature object.
 A GeoJSON object with the type "FeatureCollection" is a feature
 collection object. A feature collection object MUST have a member with
 the name "features". The value of "features" is an array. Each element
-of the array is a feature object as defined above. It is possible for
-this array to be empty.
+of the array MUST be a feature object as defined above. It is possible
+for this array to be empty. The values in the array are ordered, meaning
+that GeoJSON implementations MUST maintain order when processing them.
```

Comment:
> I can imagine different ordering being useful to different kinds of
> applications, and I'd like to be able to freely sort as I need within my
> data processing pipelines. In other words, I would flaunt that MUST in
> some parse-reserialize situations, and I think other developers would
> also flaunt it.

the question isn't whether you're free to sort. you always are, if you 
know what you're doing. the question is whether the order is part of 
what matters. you can look at it in exactly the same way as attributes 
and elements in XML:

* for attributes, it was decided that order does not matter. so most 
implementations don't even expose attribute order, and that's fine. the 
spec blessed that. some do, because some people still want to know what 
happened in the source. that's fine, too.

* for elements, if you do not expose order, or decide to freely change 
it, you're a broken implementation. and there's a spec that i can point 
to that tells you this. this doesn't mean you cannot do it, it just 
means that when you do it, be aware that you're changing things that 
applications may depend on, and you're not a generic implementation anymore.

> I'm also not aware of any JSON parsers that give access to array
> elements in other than their JSON order, so this MUST looks well
> enforced already.

yes, JSON parsers will do that for you. and that's good. the question is 
mostly what's the expected behavior for GeoJSON devs: if i write a 
GeoJSON parser and library, should i expose the order, or can i just 
expose the collection as an unordered collection? the MUST says i have 
to expose it, which is what we want, right?